libGitWrap/Private/DiffListPrivate.hpp

```diff
@@ -36,6 +36,7 @@ namespace Git
         {
         public:
             explicit DiffListPrivate(const RepositoryPrivate::Ptr& repo, git_diff* diff);
+            explicit DiffListPrivate(RepositoryPrivate* repo, git_diff* diff);
```

Comment:
Have you ever seen [Scott Meyer's talk on caches](https://vimeo.com/97337258)? Its an hour of talk that is really worth watching 2 or 3 times until it's fully understood...

However, the result is that in a mass-operation program, it _may_ be way more efficient to have the smart pointer creation outside of this code. The smart pointer code that we use ought to be in instruction cache most of the time; this secondary constructor might not. IOW: It's likely that a modern CPU will execute the technically inefficient code about 10 times faster than the optimised code.